components/partners/partnerrepository/src/main/java/org/mozilla/lite/partner/Repository.java

```diff
@@ -54,8 +53,12 @@ public void reloadData() {
         nextSubscription();
     }
 
-    public void setOnDataChangedListener(OnDataChangedListener listener) {
+    public void setOnDataChangedListener(@Nullable OnDataChangedListener listener) {
         this.onDataChangedListener = listener;
+        if (listener == null) {
```

Comment:
Setting listener to null does not always imply a cacheIsDirty. Move
```
itemPojoList = new ArrayList<>();
cacheIsDirty = true;
```
to another function call such as reset();

Reply:
will fix in another PR